x/mongo/driver/auth/internal/awsv4/signer.go

```diff
@@ -0,0 +1,468 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Based on github.com/aws/aws-sdk-go by Amazon.com, Inc.
+// See THIRD-PARTY-NOTICES for original license terms
+
+package awsv4
```

Comment:
I noticed you decided to simplify the `aws/signer/v4` API while copying it into this package. What tradeoffs did you consider when making that decision?
(I think reducing the API surface area to only what is needed is a good idea overall, I'm just curious about your decision process.)

Also, for all the Go files in the `awsv4` package, consider adding a link to the Go file in the AWS SDK that it's based on (permalink, with the commit). That will make some questions later easier to answer, like "What version of the Go AWS SDK is this based on?" and "What file should I look at to update the code in case I have to later?"